BrainPortal/app/controllers/userfiles_controller.rb

```diff
@@ -188,6 +188,19 @@ def create_parent_child #:nodoc:
     redirect_to :action => :index
   end
 
+  # Use the rest of the route as a path into a file collection.
+  # [:file_path] a path relative to the file collection's route directory.
+  #
+  # GET /userfiles/1/file_collection_content/path/to/file/in/file_collection
+  def file_collection_content
+    @userfile = Userfile.find_accessible_by_user(params[:id], current_user, :access_requested => :read)
```

Comment:
the `@userfile =` statement shoudl end with a `.first`

Reply:
I can do a check for
```
if @userfile.respond_to?(:first)
      @userfile = @userfile.first
end
```
but I should note I was going by this line: 
https://github.com/aces/cbrain/blob/master/BrainPortal/app/controllers/userfiles_controller.rb?utf8=%E2%9C%93#L203

`.first` does not seem to always be a method of the result of `Userfile.find_accessible_by_user`

Reply:
I think you are perhaps mixing up `find_accessible_by_user` with `find_all_accessible_by_user`?

Reply:
Yes you were right